Dewpoint from RH input validation.

jrleeman opened the pull request:
Add warning if relative humidity in dewpoint calculation is >120%. Does not address the divide by zero case if 0% RH is used. Is that an obvious enough message as is? Closes #424

dopplershift commented:
Ah....`log(0)` in the case of RH==0. Nothing really *to* do. That should at least give warnings of their own sort.

jrleeman commented:
With 0% RH:

```python
/Users/johnleeman/gitrepos/MetPy/metpy/calc/thermo.py:473: RuntimeWarning: divide by zero encountered in log
  val = np.log(e / sat_pressure_0c)
/Users/johnleeman/miniconda3/envs/metpydev/lib/python3.6/site-packages/pint/quantity.py:1403: RuntimeWarning: divide by zero encountered in log
  out = uf(*mobjs)
/Users/johnleeman/miniconda3/envs/metpydev/lib/python3.6/site-packages/pint/quantity.py:802: RuntimeWarning: invalid value encountered in double_scalars
  magnitude = magnitude_op(new_self._magnitude, other._magnitude)
```

dopplershift commented:
Need a `np.any` in your check...

jrleeman commented:
Yep just noticed this failed. Updating before hitting the road.

dopplershift commented:
Fix up imports (and maybe rebase on master), and this is ready to go.

jrleeman commented:
Typo with `np.any` made the warning not trip on scalar values. Fixed that and added some tests to verify we throw warnings when we should and not when we shouldn't.
